docs/conf.py

```diff
@@ -146,3 +150,84 @@
 
 # set a maximum number of days to cache remote inventories
 intersphinx_cache_limit = 0
+
+# this we'll need to build the documentation from sphinx-apidoc ourselves
+# TODO: is the actually the best way to do it? Saves us a lot of trouble but the documentation
+# cannot be changed by us if we completely rely on the ReadTheDocs backend to run sphinx-apidoc for us
```

Comment:
Can you explain what the tradeoffs are for this?

Reply:
The problem is that sphinx-apidoc is very rigid when it comes to how it builds the .md roadmaps. That's why the table of contents (aka the toctree) on the left, as well as each individual module and function name, looks something like ark.utils.spatial_analysis_utils.compute_close_cell_num.

If you allow ReadTheDocs to run sphinx-apidoc for you on their backend, you do no work outside of pushing to GitHub, but you have no way to change how sphinx-apidoc builds your documentation.

On the other hand, if we force each member of our team to be very uptight about running the Makefile locally (which includes the sphinx-apidoc call) and committing our _markdown files to the Git repo directly, then that's a lot more work for us. However, we have the advantage of changing the style of our documentation to look much prettier. Problem is, we have to do that _each_ time we run the Makefile because lo and behold, sphinx-apidoc will overwrite your old nicely-formatted documentation with its new version with ugly headings.

VVL seems to be using something like the second method as I can clearly see their .rst files in a separate folder (API). However, while they have set up a nicer table of contents system, it also looks like they got really lazy and just stopped reformatting the doc headings each time they ran sphinx-apidoc. I mean, who _really_ wants to go through those ugly-looking rst files (md files in our case) and reformat them every time?

I'm not sure how other projects do it, but I imagine some teams only run sphinx-apidoc once to build their initial version of the documentation, reformat it afterward, and then never run it again, instead, having some people add to and change the documentation manually to preserve the sanity of how their documentation displays on ReadTheDocs. It probably works really well if you don't expect to be committing a ton of new documentation each commit.

I personally don't mind our method, but that's because I can actually make sense of the whole blah.blah.blah mess. It is possible that some of the other members of the lab might find it somewhat cumbersome to have to sort through that.